### PR TITLE
fix: 151 Fixed incorrect `readVarLong` behavior in case of prematurely ended stream in `ReadableStreamingData`

### LINKED ISSUE #151: fix: Incorrect `readVarLong` behavior in case of prematurely ended stream in `ReadableStreamingData`
### Description

`ReadableStreamingData.readVarLong` behaves incorrectly in case of incomplete stream. `*ProtoCodec#parseInternal` method expects it to throw `EOFException` so that it could go to the following logic branch:
```
catch (EOFException e) {
            // Do nothing.
            // This is a workaround of hasRemaining() returning true, even there are no more elements.
            // The subsequent readByte will throw EOFException if there ar no more elements.
        }
```
Indeed, `hasRemaining` may return true in case of streams because we don't know the number of elements before we reach the end of the stream

### Steps to reproduce

1. Use PBJ 0.7.10 with hedera-services
2. Run `BlockRecordWriterV6Test.WritingTests#writingTest`
3. Observe the following exception:
```
java.nio.BufferUnderflowException
	at com.hedera.pbj.runtime.io.stream.ReadableStreamingData.readVarLong(ReadableStreamingData.java:253)
	at com.hedera.pbj.runtime.io.ReadableSequentialData.readVarInt(ReadableSequentialData.java:442)
	at com.hedera.hapi.streams.codec.RecordStreamFileProtoCodec.parseInternal(RecordStreamFileProtoCodec.java:173)
	at com.hedera.hapi.streams.codec.RecordStreamFileProtoCodec.parse(RecordStreamFileProtoCodec.java:41)
	at com.hedera.hapi.streams.codec.RecordStreamFileProtoCodec.parse(RecordStreamFileProtoCodec.java:27)
	at com.hedera.node.app.records.impl.producers.formats.v6.BlockRecordReaderV6.read(BlockRecordReaderV6.java:102)
	at com.hedera.node.app.records.impl.producers.formats.v6.BlockRecordWriterV6Test$WritingTests.writingTest(BlockRecordWriterV6Test.java:342)
```

### Additional context

_No response_

### Hedera network

other

### Version

v0.7.10

### Operating system

None

### PR DESCRIPTION
**Pull request by imalygin**
Fixes #151

### CODE REVIEW

File: pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/io/stream/ReadableStreamingDataTest.java [hunk 1]
```diff
@@ -232,4 +271,42 @@ void readFileThatDoesntExist() throws IOException {
         Files.delete(file);
         assertThrows(IOException.class, () -> new ReadableStreamingData(file));
     }
+
+    private static class TestReadeableSequentialData implements ReadableSequentialData {
```
Comment (artemananiev) [inline pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/io/stream/ReadableStreamingDataTest.java:275]: What's the purposed of this class?

File: pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/io/stream/ReadableStreamingDataTest.java [hunk 2]
```diff
@@ -153,6 +156,42 @@ public void close() throws IOException {
...
+
+    @Test
+    void incompleteStreamToBufferedData() {
+        final var inputStream = new ByteArrayInputStream(new byte[] {
+                (byte) 128, (byte) 129, (byte) 130, (byte) 131});
+
+        final var stream = new TestReadeableSequentialData(new ReadableStreamingData(inputStream));
+        stream.limit(8);
+        BufferedData buffer = BufferedData.allocate(8);
+
```
Comment (artemananiev) [inline pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/io/stream/ReadableStreamingDataTest.java:190]: Extra line

File: pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/io/stream/ReadableStreamingDataTest.java [hunk 3]
```diff
@@ -153,6 +156,42 @@ public void close() throws IOException {
...
+    }
+
+    @Test
+    void incompleteStreamToByteBuffer() {
+        final var inputStream = new ByteArrayInputStream(new byte[] {
+                (byte) 128, (byte) 129, (byte) 130, (byte) 131});
+
+        final var stream = new TestReadeableSequentialData(new ReadableStreamingData(inputStream));
+        ByteBuffer buffer = ByteBuffer.allocate(8);
+
```
Comment (artemananiev) [inline pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/io/stream/ReadableStreamingDataTest.java:177]: Extra line

File: pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/io/stream/ReadableStreamingDataTest.java (hunk 1 above)
Comment (imalygin) [reply pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/io/stream/ReadableStreamingDataTest.java:275]: The purpose is to test default implementations of
`ReadableSequentialData#readBytes(ByteBuffer)`  and `ReadableSequentialData#readBytes(BufferedData)`

Comment (anthony-swirldslabs) [reply pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/io/stream/ReadableStreamingDataTest.java:272]: Yeah, it seems to delegate all the operations to a given object.

A little javadoc would help understand why it's even needed.

File: pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/io/stream/ReadableStreamingDataTest.java (hunk 3 above)
Comment (imalygin) [reply pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/io/stream/ReadableStreamingDataTest.java:177]: Fixed

File: pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/io/stream/ReadableStreamingDataTest.java (hunk 2 above)
Comment (imalygin) [reply pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/io/stream/ReadableStreamingDataTest.java:189]: Fixed

**Review by artemananiev — APPROVED** [review]

**Review by anthony-swirldslabs — DISMISSED** [review]

File: pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/io/stream/ReadableStreamingDataTest.java (hunk 1 above)
Comment (imalygin) [reply pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/io/stream/ReadableStreamingDataTest.java:272]: Sure, let me add it

Comment (imalygin) [reply pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/io/stream/ReadableStreamingDataTest.java:272]: Done

**Review by anthony-swirldslabs — APPROVED** [review]

**Review by artemananiev — APPROVED** [review]

**Pull request merged**